shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -510,6 +510,7 @@ public static WireCommand readFrom(DataInput in, int length) throws IOException
         final WireCommandType type = WireCommandType.DATA_APPENDED;
         final UUID writerId;
         final long eventNumber;
+        final long previousEventNumber;
```

Comment:
I'm worried that we are adding another 8 bytes to every response here, consequently, I have a couple of questions related to this:

- Would it be possible to check for sequentiality in `AppendProcessor` without additional fields?
- In the case the answer to the previous question is negative, could we send the diff between `eventNumber` and previous in the form of a `byte` or at least an `int` in the case a `byte` is not sufficient? In this case, we at least send fewer bytes.

Reply:
@fpj there are 2 checks we make. One is basic sequentiality (for which `AppendProcessor` has enough data), but the more imp one (i think) is the one about missed acks which requires comparison with whats the client got in its inflight list. This previousEventNumber delta could be represented via an int since the eventCount while batching writes is an int. It'll save us 4 bytes (if you think its worth the additional complexity in the contract). But there is a corner case where int wont work which is that server default for the value is `Long.MIN_VALUE` so the delta would be a long too for the response of first write. That could be worked around too maybe using an int with a special value but that's more complexity in the contract.